docs/schema_plugins.md

```diff
@@ -1 +1,234 @@
+from nomad.metainfo import Context, MEnum, Quantity, Section, SectionProxy, SubSection
+```
+
+- `nomad_simulations`:
+
+- `numpy`:
+
+- `ArchiveSection`:
+
+- `SubSection`:
+
+-
+
+For the following examples we will use the H5MD-NOMAD file format, which is essentially a schema for molecular dynamics simulations, specified within the hdf5 file format. Imagine we have developed a parser for this file.
+
+Because the H5MD-NOMAD file is not created directly from a simulation program, there is additional metadata (beyond the program information), specifying the authorship of the file and the specifications for the generation (creator) of the H5MD schema used:
+
+    h5md
+     +-- version: Integer[2]
+     \-- author
+     |    +-- name: String[]
+     |    +-- (email: String[])
+     \-- creator
+     |    +-- name: String[]
+     |    +-- version: String[]
+     \-- program
+          +-- name: String[]
+          +-- version: String[]
+
+The program information can be stored under the existing `Program()` class within `nomad-simulations`. We already learned how to implement this into our parser:
+
+```python
+from nomad_simulations.schema_packages.general import Simulation, Program
+
+simulation = Simulation()
+simulation.program = Program(
+    name=group_h5md_dict.get('program_name'),
+    version=group_h5md_dict.get('program_version'),
+)
+```
+
+ However, we need to create a new class for the rest of the metadata. The "creator" information is referring to a secondary software used to create the hdf5 file, so it makes sense to store this in a class very similar to `Program()`. Let's first take a look at the current `Program()` schema:
+
+```python
+from nomad_simulations.general import Program
+program = Program()
+print(program.m_def.all_quantities)
+```
+
+```
+{'name': nomad_simulations.general.Program.name:Quantity,
+ 'datetime': nomad.datamodel.metainfo.basesections.BaseSection.datetime:Quantity,
+ 'lab_id': nomad.datamodel.metainfo.basesections.BaseSection.lab_id:Quantity,
+ 'description': nomad.datamodel.metainfo.basesections.BaseSection.description:Quantity,
+ 'version': nomad_simulations.general.Program.version:Quantity,
+ 'link': nomad_simulations.general.Program.link:Quantity,
+ 'version_internal': nomad_simulations.general.Program.version_internal:Quantity,
+ 'compilation_host': nomad_simulations.general.Program.compilation_host:Quantity}
+ ```
+
+ Indeed, this class contains `name` and `version` information, along with other quantities that make sense in the context of the H5MD-NOMAD creator. So, in this case we can simply reuse this class. We need to simply add a new section within the `Simulation()` class for the H5MD-NOMAD file creator that is of type `Program()`. In `schema_packages/<parser_name>_schema.py`:
+
+ ```python
+class Simulation(nomad_simulations.schema_packages.general.Simulation):
+
+    x_h5md_creator = SubSection(
+        sub_section=nomad_simulations.schema_packages.general.Program.m_def
+    )
+```
```

Comment:
This is an example of a semantic extension. If you introduced this concept in the intro, you could now point it out.